core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -790,7 +809,36 @@ class TimeSeriesShard(val ref: DatasetRef,
     shardStats.purgedPartitions.increment(numDeleted)
   }
 
-  def createFlushTask(flushGroup: FlushGroup): Task[Response] = {
+  /**
+    * Creates zero or more flush tasks (up to the number of flush groups) based on examination
+    * of the record container's ingestion time. This should be called before ingesting the container.
+    */
+  def createFlushTasks(container: RecordContainer): Seq[Task[Response]] = {
+    val tasks = new ArrayBuffer[Task[Response]]()
+
+    var oldTimestamp = lastIngestionTime
+    val ingestionTime = Math.max(oldTimestamp, container.timestamp) // monotonic clock
+    var newTimestamp = ingestionTime
+
+    if (newTimestamp > oldTimestamp && oldTimestamp != Long.MinValue) {
+      var tsAdjust = 0L
+      for (group <- 0 until numGroups optimized) {
+        if (oldTimestamp / flushBoundaryMillis != newTimestamp / flushBoundaryMillis) {
```

Comment:
I guess this is fine, though it would make a little more sense if the flushBoundaryMillis is adjusted per group instead (since it is the group's boundary that varies by group)

Reply:
The size of the division is one hour (by config), and it must remain constant.